### PR TITLE
Kubernetes support

### PR DESCRIPTION
**Pull request by jdoliner**
Adds files in `etc/kubernetes` to deploy on kubernetes.
This pull also fixes the cloud template which was broken in master.

### CODE REVIEW

File: etc/aws/template [hunk 1]
```diff
@@ -153,7 +153,7 @@
                                 "            -v /:/host:ro",
                                 "            -v /tmp/home/core:/host/home/core",
                                 "            -i pachyderm/pfs",
-                                "            /go/bin/deploy -disk /dev/xvdf ",
+                                "            /go/src/github.com/pachyderm/pachyderm/etc/bin/launch-wrapper /go/bin/deploy -disk /dev/xvdf ",
```
Comment (peter-edge) [inline etc/aws/template]: trailing space in quotes

Comment (peter-edge) [inline etc/aws/template]: or is that on purpose?

File: src/cmd/shard/main.go [hunk 2]
```diff
@@ -20,31 +21,45 @@ func main() {
...
-		"http://"+address,
-		"data-"+shardStr,
-		"comp-"+shardStr,
-		"pipe-"+shardStr,
-		shardNum,
-		modulos,
+		"http://"+*address,
+		fmt.Sprintf("data-%d-%d", *shardNum, *modulos),
+		fmt.Sprintf("pipe-%d-%d", *shardNum, *modulos),
+		uint64(*shardNum),
```
Comment (peter-edge) [inline src/cmd/shard/main.go]: i think if this is -1, this will break right?

File: src/storage/shard.go [hunk 3]
```diff
@@ -24,23 +24,25 @@ const (
...
-	cache              etcache.Cache
+	url            string
+	dataRepo       string
+	pipelinePrefix string
+	shard, modulos uint64
+	shardStr       string
+	runners        map[string]*pipeline.Runner
+	guard          *sync.Mutex
+	cache          etcache.Cache
+	roles          []string
```
Comment (peter-edge) [inline src/storage/shard.go]: you don't use this

File: etc/aws/template (hunk 1 above)
Comment (jdoliner) [inline etc/aws/template]: That's on purpose. AWS turns the next line in to a set of user defined args so we need a space there to make sure they get parsed correctly.

File: src/storage/shard.go (hunk 3 above)
Comment (jdoliner) [inline src/storage/shard.go]: Good point, removing.

File: src/cmd/shard/main.go (hunk 2 above)
Comment (jdoliner) [inline src/cmd/shard/main.go]: Hmm, so this overflows but I don't think that actually leads to an incorrect behavior. In the case when `*shardNum == -1` we run `FindRole` which sets it to a real value. It is a bit weird though. How much does it bug you?

Comment (peter-edge) [inline src/cmd/shard/main.go]: ah whatever, it bugs me, dont want guaranteed overflow situations committed, but i think we'll clean this up anyways and if it works, all good

**Pull request merged**